### DISCUSSION TITLE
How to work with AAD groups?

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [pnp/pnpcore#320] [Q&A] [answered]
When I used Graph SDK, I used to get a group by mail-nickname like so:
`
await graphClient.Groups.Request().Filter($"mailNickname eq '{myGroup.ADGroup}'").GetAsync();
`
I am trying to do the same with PnPCore but can't figure out how to do it.

```
 using (var pnpContext = await pnpContextFactory.CreateAsync("Default"))
            {
                 var group = await pnpContext.Group.GetAsync(g => g.MailNickname == projGroup.ProjectNumber );
                Console.WriteLine(group.DisplayName);
            }

```
Returns an error "PnP.Core: Unresolved tokens found in API call groups/{GraphId}" which leads me to believe I needed to create the context with the group ID like in the documentation.
But How will I get the Group's Guid if I only have the mail-nickname to work with?

Is there a way to use the pnpContext.GraphClient to run Graph queries ?

Thanks

**Comment by jansenbe** [comment, marked as answer]
@Ofer-Gal : at this moment that's not a supported scenario, PnP Core SDK works by initializing a PnPContext from either a given site url or group id. So if you have the group id 

```csharp
// Your group id
Guid groupId = new Guid("593DB9B2-2167-45FC-8387-A2438696AB8E");

using (var pnpContext = await pnpContextFactory.CreateAsync(groupId))
{
   // Work with the SharePoint site (and Team) connected to this Microsoft 365 group  
}
```

**Reply by Ofer-Gal** [reply]
Will it ever be a supported scenario? 
What about a way to use the pnpContext.GraphClient to run Graph queries ?

**Reply by jansenbe** [reply]
We to plan to offer more scenarios, having a generic Graph/Rest client exposed is not something we've considered so far, but I see how it can help in certain scenarios. I'll an issue to track this as a backlog item.
